h/views/api/bulk.py

```diff
@@ -30,13 +32,60 @@ def from_byte_stream(cls, byte_stream, executor, observer=None):
             return None
 
 
+class AuthorityCheckingExecutor(AutomaticReportExecutor):
+    """A bulk executor which checks the authority."""
+
+    def __init__(self, authority="lms.hypothes.is"):
+        self.effective_user = None
+        self.authority = authority
+
+    def configure(self, config):
+        self._assert_authority("effective user", config.effective_user)
+
+        self.effective_user = config.effective_user
+
+    def execute_batch(self, command_type, data_type, default_config, batch):
+        for command in batch:
+            self._check_authority(data_type, command.body)
+
+        return super().execute_batch(command_type, data_type, default_config, batch)
+
+    def _assert_authority(self, field, value, embedded=True):
+        if embedded and value.endswith(f"@{self.authority}"):
+            return
+
+        if value == self.authority:
+            return
+
+        raise InvalidDeclarationError(
+            f"The {field} '{value}' does not match the expected authority"
+        )
+
+    def _check_authority(self, data_type, body):
+        if data_type == DataType.USER:
```

Comment:
Should these be calls on the command objects? They know where everything lives, they could be in charge of this?